Sources/Foundation/ProcessInfo.swift

```diff
@@ -80,11 +80,68 @@ open class ProcessInfo: NSObject {
         return CFUUIDCreateString(kCFAllocatorSystemDefault, uuid)._swiftObject
     }
 
+#if os(Windows)
+    internal var _rawOperatingSystemVersionInfo: RTL_OSVERSIONINFOEXW? {
+        guard let ntdll = ("ntdll.dll".withCString(encodedAs: UTF16.self) {
+            LoadLibraryExW($0, nil, DWORD(LOAD_LIBRARY_SEARCH_SYSTEM32))
+        }) else {
+            return nil
+        }
+        defer { FreeLibrary(ntdll) }
+        typealias RTLGetVersionTy = @convention(c) (UnsafeMutablePointer<RTL_OSVERSIONINFOEXW>) -> NTSTATUS
+        guard let pfnRTLGetVersion = unsafeBitCast(GetProcAddress(ntdll, "RtlGetVersion"), to: Optional<RTLGetVersionTy>.self) else {
+            return nil
+        }
+        var osVersionInfo = RTL_OSVERSIONINFOEXW()
+        osVersionInfo.dwOSVersionInfoSize = DWORD(MemoryLayout<RTL_OSVERSIONINFOEXW>.size)
+        guard pfnRTLGetVersion(&osVersionInfo) == 0 else {
+            return nil
+        }
+        return osVersionInfo
+    }
+#endif
+
     open var operatingSystemVersionString: String {
         let fallback = "Unknown"
 #if os(Linux)
-        let version = try? String(contentsOf: URL(fileURLWithPath: "/proc/version_signature", isDirectory: false), encoding: .utf8)
-        return version ?? fallback
+        var utsNameBuffer = utsname()
+        guard uname(&utsNameBuffer) == 0 else {
+            return fallback
+        }
+        let release = withUnsafePointer(to: &utsNameBuffer.release.0) { String(cString: $0) }
```

Comment:
Is `uname -r` really the right thing for Linux? Just as we return the macOS/iOS version on Apple platforms rather than the Darwin version, perhaps we should return the distribution version instead of the Linux version? (i.e. /etc/lsb_release, /etc/debian_version and so on)

Here's the approach we took in Qt: https://github.com/qt/qtbase/blob/dev/src/corelib/global/qglobal.cpp -- look for findUnixOsVersion and start from there.

Reply:
The `/etc/*-release` is far more free form.  Not everyone really bothers with LSB, so, at least Gentoo, exherbo, Fedora, and debian have `/etc/{distro}-release`.  Its also interesting for distributions which are rolling distributions and thus have no real version number associated with them.  So although your stage may be versioned, the state can vary drastically between installations.

Reply:
My primary aim with this change was to provide _some_ kind of remotely usable value - it's not usually meaningful to make decisions based on a Linux environment's version in a generic sense (i.e. when not limited to a specific distro) anyway, so accuracy wasn't a primary concern; it was more about addressing that the `/proc/version_signature` approach yields nothing at all on most current Linuxes. I could certainly write the code to loop over the set of, say, `/etc/os-release`, `/etc/system-release`, `/etc/lsb-release`, `/etc/debian_version`, `/etc/redhat-release`, `/etc/alpine-release`, `/etc/centos-release`, and so forth (probably in more or less that order) - just `os-release` by itself would already cover everything supported by Swift 5. The question to me is, does doing so confer useful enough information to be worth even that much bother versus just letting `uname*(` give a number?

Reply:
I dont have a particularly strong view on this but here's some thoughts:

1. I see `operatingSystemVersionString` as being a stringification of `operatingSystemVersion` - which on Linux is the kernel version - so on the one hand it should just be that and say `Version 5.32.0` or whatever. Note that on macOS today it says something like `Version 10.14.6 (Build 18G5033)`. The `operatingSystemVersion` tuple is actually useful on Linux for checking the availability of system calls.

2. However it should probably display something more than just the kernel version and most users would probably be confused by seeing `Linux 5.2.32` rather than `Ubuntu 20.04`. So I think the code should do a best effort to try and find the distribution name/version and fallback to the generic `Linux <kernel_version>` as a last resort (and make sure this is cached in an ivar). I think we want to avoid this being a User-Agent type thing where it gets parsed and programs become reliant on the format.
 
3. I think a better long term solution for Linux, and probably Windows, is to pitch to add something into the new [System](https://developer.apple.com/documentation/system) module that can return a more detailed answer including enums for the distribution etc so that developers move away from this property.

Reply:
@spevans I concur on all counts; I'll add basic best-effort distro detection with the uname fallback for now. The long-term concern is out of scope for this PR, though; this is just hoping to provide parity with yet another long-standing Foundation API (and I **do** mean long-standing - this API sits right next to `operatingSystem()` and `operatingSystemName()`, of all things! 🤣). What is the best path for making such a pitch for the `System` module?

Reply:
@gwynne I dont know how the `System` module is going to be used, Ive only seen the File stuff for it in the documentation. Maybe write a standalone package to prototype some ideas especially with enumerating the different versions of Linux and Windows.

Getting back to this PR, are there any tests that can be written to check the functionally taking into account that the tests could be run on various Linux/Windows versions? Maybe doing a simple check that one of the supported OSes (Linux/Ubuntu/Debain/Centos etc) at least appears in the result ?